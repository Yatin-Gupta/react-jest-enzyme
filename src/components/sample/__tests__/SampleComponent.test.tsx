import * as React from 'react';
import { shallow } from 'enzyme';
import SampleComponent from '../SampleComponent';
import * as renderer from 'react-test-renderer';

describe('Counter Component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<SampleComponent />);
    // @ts-ignore
    const countState = wrapper.state().counter;
    expect(countState).toEqual(0);
  });

  it('Can increment the count when the button is clicked', () => {
    const wrapper = shallow(<SampleComponent />);
    // .first() can also be used
    // or we can also speify it as: button.first
    const incrementBtn = wrapper.find('button').last();
    incrementBtn.simulate('click');
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('-2');
  });

  // Snapshot Testing
  it('SampleComponent Snapshot Testing', () => {
    const tree = renderer.create(<SampleComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
