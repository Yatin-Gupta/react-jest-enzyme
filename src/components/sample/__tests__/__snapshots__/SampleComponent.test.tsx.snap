// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component SampleComponent Snapshot Testing 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Increment Counter
  </button>,
  <button
    onClick={[Function]}
  >
    Decrement Counter
  </button>,
  <p>
    0
  </p>,
]
`;
